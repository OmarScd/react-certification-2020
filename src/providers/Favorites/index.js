export { default } from './Favorites.provider';
