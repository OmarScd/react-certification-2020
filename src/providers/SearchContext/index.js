export { default, useSearch } from './SearchContext.provider';
