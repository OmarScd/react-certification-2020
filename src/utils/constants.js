const USERNAME_STORAGE_KEY = 'osa-reactcert-2020-username';

export { USERNAME_STORAGE_KEY };
