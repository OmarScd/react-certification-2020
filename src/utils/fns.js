// TODO: convert to fetch
const getQueryURL = (query) =>
  `https://content.googleapis.com/youtube/v3/search?part=id&part=snippet&q=${query}&maxResults=25&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`;

const getRelatedURL = (videoId) =>
  `https://www.googleapis.com/youtube/v3/search?part=snippet&part=id&relatedToVideoId=${videoId}&type=video&maxResults=25&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`;

export { getQueryURL, getRelatedURL };
