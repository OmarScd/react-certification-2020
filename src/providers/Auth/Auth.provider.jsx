import React, { useState, useEffect, useContext } from 'react';

import { USERNAME_STORAGE_KEY } from '../../utils/constants';
import { storage } from '../../utils/storage';

const AuthContext = React.createContext(null);

const useAuth = () => {
  const context = useContext(AuthContext);
  if (!context) {
    throw new Error(`Can't use "useAuth" without an AuthProvider!`);
  }
  return context;
};

const AuthProvider = ({ children }) => {
  const [authenticated, setAuthenticated] = useState(false);
  const [username, setUsername] = useState(undefined);

  useEffect(() => {
    const user = storage.get(USERNAME_STORAGE_KEY);
    const isAuthenticated = !!user;

    setAuthenticated(isAuthenticated);
    if (isAuthenticated) {
      setUsername(user);
    }
  }, []);

  const login = (name) => {
    setAuthenticated(true);
    setUsername(name);
    storage.set(USERNAME_STORAGE_KEY, name);
  };

  const logout = () => {
    setAuthenticated(false);
    setUsername(undefined);
    storage.remove(USERNAME_STORAGE_KEY);
  };

  return (
    <AuthContext.Provider value={{ login, logout, authenticated, username }}>
      {children}
    </AuthContext.Provider>
  );
};

export { useAuth, AuthProvider as default };
