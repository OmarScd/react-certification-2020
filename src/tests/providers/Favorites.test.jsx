import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FavoritesProvider from '../../providers/Favorites';

import '../../utils/storage';
import '../../providers/Auth';

jest.mock('../../providers/Auth', () => ({
  default: jest.fn(),
  useAuth: () => ({ authenticated: true }),
}));

jest.mock('../../utils/storage', () => {
  return {
    storage: {
      get: jest.fn(),
      set: jest.fn(),
      remove: jest.fn(),
    },
  };
});

describe('FavoritesProvider', () => {
  afterAll(cleanup);

  it('Renders correctly', () => {
    const { container } = render(
      <FavoritesProvider>
        <p>Mock content</p>
      </FavoritesProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
