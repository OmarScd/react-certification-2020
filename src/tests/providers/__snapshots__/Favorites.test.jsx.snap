// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesProvider Renders correctly 1`] = `
<p>
  Mock content
</p>
`;
